Tests/Integration/MobileCoinClientPublicApiIntTests.swift

```diff
@@ -1,573 +1,821 @@
-
-            if let amountPicoMob = try? XCTUnwrap(client.balance.amountPicoMob()) {
-                print("balance: \(amountPicoMob)")
-                XCTAssertGreaterThan(amountPicoMob, 0)
-            }
-
-            expect.fulfill()
+        try await client.updateBalances()
+        if let amountPicoMob = try? XCTUnwrap(client.balance(for: .MOB).amount()) {
+            print("balance: \(amountPicoMob)")
+            XCTAssertGreaterThan(amountPicoMob, 0)
         }
     }
 
-    func testPrintBalances() throws {
+    func testPrintBalances() async throws {
+        try XCTSkip()
         let description = "Printing account balance"
-        try testSupportedProtocols(description: description) {
-            try printBalances(transportProtocol: $0, expectation: $1)
+        try await testSupportedProtocols(description: description) {
+            try await self.printBalance(transportProtocol: $0)
         }
     }
 
-    func printBalances(
-        transportProtocol: TransportProtocol,
-        expectation expect: XCTestExpectation
-    ) throws {
-        let numAccounts = IntegrationTestFixtures.testAccountCount
-        var balanceExpects: [XCTestExpectation] = []
-        try Array(0..<numAccounts).forEach({ index in
+    func printBalance(
+        transportProtocol: TransportProtocol
+    ) async throws {
+        for index in Array(0...9) {
             let key = try IntegrationTestFixtures.createAccountKey(accountIndex: index)
 
             let client = try IntegrationTestFixtures.createMobileCoinClient(
                 accountKey: key,
                 transportProtocol: transportProtocol)
 
-            let expect = XCTestExpectation(description: "Retrieving balance for acct idx \(index)")
-            client.updateBalances {
-                guard $0.successOrFulfill(expectation: expect) != nil else { return }
-
-                print("account index \(index) public address \(key.publicAddress)")
+            try await client.updateBalances()
 
-                if let amtMob = try? XCTUnwrap(client.balance(for: .MOB)) {
-                    print("account index \(index) balance: \(amtMob)")
-                }
-
-                if let amtMob = try? XCTUnwrap(client.balance(for: .MOBUSD)) {
-                    print("account index \(index) balance: \(amtMob)")
-                }
+            print("account index \(index) public address \(key.publicAddress)")
 
-                if let amtEUsd = try? XCTUnwrap(client.balance(for: .TestToken)) {
-                    print("account index \(index) balance: \(amtEUsd)")
-                }
+            if let amtMob = try? XCTUnwrap(client.balance(for: .MOB)) {
+                print("account index \(index) balance: \(amtMob)")
             }
 
-            balanceExpects.append(expect)
-        })
-
-        XCTWaiter().wait(for: balanceExpects, timeout: 40)
-        expect.fulfill()
-    }
-
-    func testBalances() throws {
```

Comment:
did this test make it into your additions ?

Reply:
Added an async version back in - tx